src/server/scripts/Northrend/VaultOfArchavon/boss_archavon.cpp

```diff
@@ -107,15 +107,17 @@ class boss_archavon : public CreatureScript
                             events.ScheduleEvent(EVENT_ROCK_SHARDS, 15000);
                             break;
                         case EVENT_CHOKING_CLOUD:
-                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, [&](Unit* u) { return u && !(u->GetDistance(me->GetPosition()) < 10.0f); }))
+                            {
                                 DoCast(target, SPELL_CRUSHING_LEAP, true); //10y~80y, ignore range
+                                Talk(EMOTE_LEAP, target);
```

Comment:
-> spell hit handler

Reply:
Isn't it Unneccesary amount of code?
And doesn't it announce the spell at incorrect time this way too

Reply:
> Tested ingame, does emote during the leap instead of after it.

I thought he is intended to do the emote after the leap. If he should do it during the leap, then it's fine.